### ISSUE TITLE
Accessibility: Not possible to create a new file using keyboard shortcuts from file explorer

### ISSUE THREAD

**Issue by auchenberg** [Microsoft/vscode#51494]
Learnings from NVAccess testing:

I can get to the new   File button. However, once pressing it, I am thrown back to the treeview,   with a new treeview item with no name. It is not clear that I have to type a   name here. The role of the object should be edit, not treeview item. Or,  focus should land on an edit field inside   the treeview item. It looks like there is a text node with the editable state   inside the treeview item, but focus is never placed there.

**Comment by isidorn** [comment]
I can not reproduce this.
When I tab into the `new file` action and press it focus is in the input box (which you call edit field).
The input box has the following aria label:
`aria-label="Type file name. Press Enter to confirm or Escape to cancel."`

The focus is not in the treeview item.

Can you reprodiuce this with latest vscode insiders?

**Comment by auchenberg** [comment]
Re-opening as I have requested more info from the original reporter, NVAccess.

**Comment by isidorn** [comment]
Sure, just paste here the info once you have it

**Comment by AccessibilityTestingTeam-TCS** [comment]
![fileexplorernewfileissue](https://user-images.githubusercontent.com/39204909/42556980-df6a0a14-850a-11e8-82f8-af979b27d0c2.png)

Focus does land on input box field but appropriate label is not read along with it. NVDA reads it as treeview item. 

Version : VSCode 1.25.0

**Comment by isidorn** [comment]
@AccessibilityTestingTeam-TCS thanks for clarifying. The issue here is that the `aria-label` is not respected on my focused element inside the tree.
@bpasero @joaomoreno do we already have some issue like this that the aria-label inside trees are not respected and only 'treeview item' is read out?

**Comment by joaomoreno** [comment]
Not sure... :thinking:

**Comment by bpasero** [comment]
Yeah: https://github.com/Microsoft/vscode/issues/52682

**Comment by isidorn** [comment]
So I guess this is also dependent on electron update. Adding upstream, sending to backlog

**Comment by isidorn** [comment]
Closing since we have updated to electron 3.0
